src/mbgl/gl/context.cpp

```diff
@@ -124,8 +124,11 @@ void Context::initializeExtensions(const std::function<gl::ProcAddress(const cha
             return nullptr;
         };
 
-        const std::string renderer = reinterpret_cast<const char*>(MBGL_CHECK_ERROR(glGetString(GL_RENDERER)));
-        Log::Info(Event::General, "GPU Identifier: %s", renderer.c_str());
+        static const std::string renderer = []() {
+            std::string r = reinterpret_cast<const char*>(MBGL_CHECK_ERROR(glGetString(GL_RENDERER)));
+            Log::Info(Event::General, "GPU Identifier: %s", r.c_str());
+            return r;
+        }();
```

Comment:
Caching this statically means that a program that creates a map, destroys it, _switches GPU_, then creates a map again will report the wrong renderer. This kind of behavior is possible on devices that use multiple GPUs, e.g. MacBooks.